gitlab/v4/objects/projects.py

```diff
@@ -894,6 +894,65 @@ def remote_import(
             "/projects/remote-import", post_data=data, **kwargs
         )
 
+    def remote_import_s3(
+        self,
+        region: str,
+        bucket_name: str,
+        file_key: str,
+        access_key_id: str,
+        secret_access_key: str,
+        path: str,
+        name: Optional[str] = None,
+        namespace: Optional[str] = None,
+        overwrite: bool = False,
+        override_params: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> Union[Dict[str, Any], requests.Response]:
+        """Import a project from an archive file stored on AWS S3.
+
+        Args:
+            region: AWS S3 region name where the file is stored
+            bucket_name: AWS S3 bucket name where the file is stored
+            file_key: AWS S3 file key to identify the file.
+            access_key_id: AWS S3 access key ID.
+            secret_access_key: AWS S3 secret access key.
+            path: Name and path for the new project
+            name: The name of the project to import. If not provided,
+                defaults to the path of the project.
+            namespace: The ID or path of the namespace that the project
+                will be imported to
+            overwrite: If True overwrite an existing project with the
+                same path
+            override_params: Set the specific settings for the project
+            **kwargs: Extra options to send to the server (e.g. sudo)
+
+        Raises:
+            GitlabAuthenticationError: If authentication is not correct
+            GitlabListError: If the server failed to perform the request
```

Comment:
Before we forget - this would technically be the way to go, but would also require a decorator `@exc.on_http_error(exc.GitlabImportError)`:

```suggestion
            GitlabImportError: If the server failed to perform the request
```

Reply:
Yup can do, hopefully `GItlabImportError` is defined beforehand

Reply:
Done, updated tests to check for this error as well